tests/interpolation/test_interpolation.py

```diff
@@ -257,7 +261,7 @@ def test_none_value_in_quoted_string(restore_resolvers: Any) -> None:
             User(name="Bond", age=SI("${cast:int,'7'}")),
             "age",
             7,
-            IntegerNode,
+            InterpolationResultNode,
```

Comment:
can you add 
```python
assert type(val) is type(expected_value)
```
to ensure the type of the value is the right one and this is not some `__eq__` magic?

Reply:
Done in f764f5c (actually had to update the DictConfig/ListConfig tests that were relying on `__eq__` magic)